interfaces/automated-testing/rid/injection.yaml

```diff
@@ -0,0 +1,180 @@
+      security:
+        - TestAuth:
+            - rid.inject_test_data
+components:
+  schemas:
+    CreateTestParameters:
+      description: 'A complete set of data to be injected into a Service Provider system under test.'
+      required:
+        - requested_flights
+      type: object
+      properties:
+        requested_flights:
+          description: >-
+            One or more logical flights, each containing test data to inject into the system.
+            Elements should be sorted in ascending order of `timestamp`.
+          type: array
+          items:
+            $ref: '#/components/schemas/TestFlight'
+    ChangeTestResponse:
+      description: ''
+      required:
+        - injected_flights
+        - version
+      type: object
+      properties:
+        injected_flights:
+          description: 'The complete set of test data actually injected into the Service Provider system under test.'
+          type: array
+          items:
+            $ref: '#/components/schemas/TestFlight'
+        version:
+          description: Version of test.  Used to delete test.
+          type: string
+          example: 1
+    DeleteTestResponse:
+      description: ''
+      required:
+        - injected_flights
+      type: object
+      properties:
+        injected_flights:
+          description: 'The complete set of test data deleted.'
+          type: array
+          items:
+            $ref: '#/components/schemas/TestFlight'
+    TestFlight:
+      description: 'The set of data to be injected into a Service Provider system under test for a single flight.'
+      required:
+        - injection_id
+        - telemetry
+        - details_responses
+      type: object
+      properties:
+        injection_id:
+          description: 'ID of the injected test flight.  Remains the same regardless of the flight ID/UTM ID reported in the system.'
+          type: string
+          example: 'edb7695f-8737-4b9f-91f8-e2afbb333f41'
+        telemetry:
+          description: >-
+            The set of telemetry data that should be injected into the system for this flight.
+            Each element follows the RIDAircraftState schema from the ASTM remote ID standard.
+          type: array
+          items:
+            $ref: 'https://raw.githubusercontent.com/uastech/standards/astm_rid_1.0/remoteid/canonical.yaml#/components/schemas/RIDAircraftState'
+        details_responses:
+          description: The details of the flight as a function of time.
+          type: array
+          items:
+            $ref: '#/components/schemas/TestFlightDetails'
+    TestFlightDetails:
+      description: The set of data with which the Service Provider system under test should respond when queried for the details of a test flight.
+      required:
+        - effective_after
+        - details
+      type: object
+      properties:
+        effective_after:
```

Comment:
At the moment, are we expecting that the track data will be submitted right after the tracks are generated? This is simpler and will work. The other option is to have a `test_start` and `test_end` parameters so we can say tracks / test data is created today for testing tomorrow (for e..g.)

Reply:
Eventually, the tracks should be generated once and then committed to the repository so they can be audited (we'll do this once we get good tracks that have been verified to produce good test results).  When playing those tracks back, I'm imagining the replayer will probably leave a minute or so between the time the test data is delivered to a USS and the times the telemetry points should be applicable.  This allows for additional time on the automated testing side to announce test data to other USSs, and it allows time on the USS side to initialize its system with the injected data.

Currently, all injected data has/should have timestamps associated with it, so hypothetically rid_qualifier could provide it to the injector a day before the test.  Though I don't think that is a likely scenario as I wouldn't expect injected data to necessarily persist through a system restart, so if a USS redeployed their system, they may lose the data intended for the next day.

Reply:
OK got it. I think then we can drop the `test_start` and `test_end` parameters. I agree that when we have robust track data and track types we can commit it to the repository and at the moment we will test and develop these starting with what we have.